data-processor/src/main/java/io/micronaut/data/processor/visitors/RepositoryTypeElementVisitor.java

```diff
@@ -515,6 +518,25 @@ private void bindParameters(boolean supportsImplicitQueries,
                 if (!supportsImplicitQueries) {
                     builder.member(DataMethodQueryParameter.META_MEMBER_NAME, p.getName());
                 }
+                Object value = p.getValue();
+                if (value != null) {
+                    if (value instanceof String expression) {
+                        // TODO: Support adding an expression annotation value in Core
```

Comment:
can you create an issue for this in core and link here?

Reply:
https://github.com/micronaut-projects/micronaut-core/issues/10734